scripts/windows/restart-helpers.ps1

```diff
@@ -0,0 +1,53 @@
+# //////////////////////////////////////////////////////////////////////
+# restart-helpers.ps1
+
+function Test-Key([string] $path, [string] $key)
+{
+  return ((Test-Path $path) -and ((Get-Key $path $key) -ne $null))
+}
+
+function Remove-Key([string] $path, [string] $key)
+{
+  Remove-ItemProperty -path $path -name $key
+}
+
+function Set-Key([string] $path, [string] $key, [string] $value)
+{
+  Set-ItemProperty -path $path -name $key -value $value
+}
+
+function Get-Key([string] $path, [string] $key)
+{
+  return (Get-ItemProperty $path).$key
+}
+
+
+$global:restartKey = "Restart-And-Resume"
+function Clear-Any-Restart([string] $key=$global:restartKey)
+{
+  if (Test-Key $global:RegRunKey $key) {
+    Remove-Key $global:RegRunKey $key
+  }
+}
+
+$global:RegRunKey ="HKLM:\SOFTWARE\Microsoft\Windows\CurrentVersion\Run"
+$global:powershell = (Join-Path $env:windir "system32\WindowsPowerShell\v1.0\powershell.exe")
+function Restart-Machine-With-Resume-Command([string] $script, [string] $step, [string] $action)
```

Comment:
Maybe the callers should be more explicit about the parameters, and not use them positionally? That is, do `Restart-Machine-With-Resume-Command -Script script.ps1 -Step …`

(Actually, isn't the `-Script` constant and relative to this file, and therefore doesn't even need to be given?)

Reply:
This feels too quibbly. The code's working and it's clear enough what's going on. And it is a sort of shell-script language.